### PR TITLE
fix typo of predicate arguments in Get route methods

### PR DESCRIPTION
**Pull request by Nipodemos**
Just noticed in #417 that i documented wrongly the predicate argument in all methods that use it. Fixing.

### CODE REVIEW

File: lib/src/get_main.dart [hunk 1]
```diff
@@ -154,17 +154,19 @@ class GetImpl implements GetService {
...
   }
 
-  /// Push the given [page], and then pop several [pages] in the stack until
+  /// Push the given [page], and then pop several pages in the stack until
   /// [predicate] returns true
   ///
   /// [id] is for when you are using nested navigation,
   /// as explained in documentation
   ///
+  /// TODO: Learn how to get a Route class to use with this method
```
Comment (Nipodemos) [inline lib/src/get_main.dart:163]: Also, i've put a todo in docs, i don't know how would i get a Route method in Get, can you explain that to me?

File: lib/src/get_main.dart [hunk 2]
```diff
@@ -144,7 +144,7 @@ class GetImpl implements GetService {
   /// as explained in documentation
   ///
   /// [predicate] can be used like this:
-  /// `Get.until(Get.currentRoute == '/home')`so when you get to home page,
+  /// `Get.until((route) => Get.currentRoute == '/home')`so when you get to home page,
```
Comment (jonataslaw) [inline lib/src/get_main.dart:147]: Nice!

File: lib/src/get_main.dart (hunk 1 above)
Comment (jonataslaw) [reply lib/src/get_main.dart:163]: In fact, he doesn't need a route class, just a VoidCallback where he delivers the route and gets a response bool.

(route)=> for example, route is given by callback, so you can do something like:

Get.until((route) => route.isFirst)

Comment (Nipodemos) [reply lib/src/get_main.dart:163]: but that is not the predicate argument?

`Get.until()` receives as first argument predicate, which you told me once is exactaly that, a VoidCallback

but `Get.offUntil()` has a `Route<T> page` and also the `RoutePredicate predicate`
i have doubts abou the first one

**Comment by Nipodemos** [comment]
hey i think you have forgot me kkkkkk

i still have a doubt that i replied on that piece of code, let me replicate here:

> but that is not the predicate argument?
> 
> `Get.until()` receives as first argument predicate, which you told me once is exactaly that, a VoidCallback
> 
> but `Get.offUntil()` has a `Route<T> page` and also the `RoutePredicate predicate`
i have doubts abou the first one

**Comment by jonataslaw** [comment]
> hey i think you have forgot me kkkkkk
> 
> i still have a doubt that i replied on that piece of code, let me replicate here:
> 
> > but that is not the predicate argument?
> > `Get.until()` receives as first argument predicate, which you told me once is exactaly that, a VoidCallback
> > but `Get.offUntil()` has a `Route<T> page` and also the `RoutePredicate predicate`
> > i have doubts abou the first one

Predicate is a function that performs a condition when false, and does not execute it when true. It has a VoidCallback that will receive a Route<T>.

You can for example use (_) => false as a predicate so that this function is executed whenever the loop is started. Get.offAll for example, used this before version 1.17.

Get.offUntil() will close the page, and proceed to the next Route<T>.
It will do this shortly after closing all pages according to the predicate. Imagine that you have a testing app. You entered the home, clicked on math, and started the exam. You want that when the user finishes the last question, he moves to the score / note page, but you want all pages of the exam to be removed from the stack (because it makes no sense to allow the user to go back to the questions if he has already finished the exam).
Then, you will go to the result page (Route) and close all pages until home ((route) => Get.currentRoute == '/ home').
If you did not understand the operation, I will explain in Portuguese, it may be easier:


///////////////////// PORTUGUESE VERSION:

Predicado é uma função que executa uma condição quando falsa, e não a executa quando verdadeira. Ela tem um VoidCallback que irrá receber uma Route<T>. 

Você pode por exemplo usar (_)=> false como predicado para que essa função seja executada sempre que o loop for iniciado. Get.offAll por exemplo, usava isso antes da versão 1.17. 

Get.offUntil() irá fechar a página, e avançar para a próxima Route<T>. 
Ela irá fazer isso, logo após, fechar todas as páginas de acordo com o predicado. Imagine que você tem um app de testes. Você entrou na home, clicou em matemática, e iniciou o exame. Você quer que quando o usuário termine a última questão, ele avance para a página de pontuação/nota, mas você quer que todas páginas do exame sejam removidas da stack (porque não faz sentido permitir o usuário que volte para as questões se ele já terminou o exame).
Logo, você irá ir para página do resultado (Route) e irá fechar todas as páginas até a home ( (route)=> Get.currentRoute == '/home' ).

**Comment by Nipodemos** [comment]
what i want to make sure is the `page` argument, because it is not of type Widget, how should a pass a widget to `Get.offUntil()`? @jonataslaw

**Comment by jonataslaw** [comment]
> what i want to make sure is the `page` argument, because it is not of type Widget, how should a pass a widget to `Get.offUntil()`? @jonataslaw

GetPageRoute(page:()=> NextPage())

**Comment by jonataslaw** [comment]
Maybe it would be easier to copy the code from Get.to, and send a Widget and use parameters, but that would be a break update, so it would not be possible to do this in version 3.

**Comment by Nipodemos** [comment]
> Maybe it would be easier to copy the code from Get.to, and send a Widget and use parameters, but that would be a break update, so it would not be possible to do this in version 3.

makes sense, i'm fine with that.



> > what i want to make sure is the `page` argument, because it is not of type Widget, how should a pass a widget to `Get.offUntil()`? @jonataslaw
> 
> GetPageRoute(page:()=> NextPage())

Thank you very much for the explanation, it was exactly what i wanted. I'll update the code and fix the conflicts

**Comment by Nipodemos** [comment]
oh gosh everything is in a differente place, i'll have to make another PR

**Pull request closed without merging**